python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -24,457 +26,360 @@ class RouterBgpMixin(Protocol):
+    def _set_address_family_ipv4(self: AvdStructuredConfigOverlayProtocol) -> None:
         """Deactivate the relevant peer_groups in address_family_ipv4."""
-        peer_groups = []
-
+        peer_groups = self.structured_config.router_bgp.address_family_ipv4.peer_groups
         if self.shared_utils.is_wan_router:
-            peer_groups.append({"name": self.inputs.bgp_peer_groups.wan_overlay_peers.name, "activate": False})
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.wan_overlay_peers.name, activate=False)
             if self._is_wan_server_with_peers:
-                peer_groups.append({"name": self.inputs.bgp_peer_groups.wan_rr_overlay_peers.name, "activate": False})
+                peer_groups.append_new(name=self.inputs.bgp_peer_groups.wan_rr_overlay_peers.name, activate=False)
 
         # TODO: no elif
         elif self.shared_utils.overlay_evpn_vxlan is True:
-            peer_groups.append({"name": self.inputs.bgp_peer_groups.evpn_overlay_peers.name, "activate": False})
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.evpn_overlay_peers.name, activate=False)
 
         if self.shared_utils.overlay_routing_protocol == "ebgp" and (
             self.shared_utils.node_config.evpn_gateway.evpn_l2.enabled or self.shared_utils.node_config.evpn_gateway.evpn_l3.enabled
         ):
-            peer_groups.append({"name": self.inputs.bgp_peer_groups.evpn_overlay_core.name, "activate": False})
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.evpn_overlay_core.name, activate=False)
 
         if self.shared_utils.overlay_routing_protocol == "ibgp":
             if self.shared_utils.overlay_mpls is True:
-                peer_groups.append({"name": self.inputs.bgp_peer_groups.mpls_overlay_peers.name, "activate": False})
+                peer_groups.append_new(name=self.inputs.bgp_peer_groups.mpls_overlay_peers.name, activate=False)
 
             if self._is_mpls_server is True:
-                peer_groups.append({"name": self.inputs.bgp_peer_groups.rr_overlay_peers.name, "activate": False})
+                peer_groups.append_new(name=self.inputs.bgp_peer_groups.rr_overlay_peers.name, activate=False)
 
         if self.shared_utils.overlay_ipvpn_gateway is True:
-            peer_groups.append({"name": self.inputs.bgp_peer_groups.ipvpn_gateway_peers.name, "activate": False})
-
-        return {"peer_groups": peer_groups}
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.ipvpn_gateway_peers.name, activate=False)
 
-    def _address_family_evpn(self: AvdStructuredConfigOverlayProtocol) -> dict | None:
-        address_family_evpn = {}
-
-        peer_groups = []
-
-        overlay_peer_group = {}
+    def _set_address_family_evpn(self: AvdStructuredConfigOverlayProtocol) -> None:
+        peer_groups = self.structured_config.router_bgp.address_family_evpn.peer_groups
+        overlay_peer_group = EosCliConfigGen.RouterBgp.AddressFamilyEvpn.PeerGroupsItem()
```

Comment:
Let's move this one down below the wan-route section (so ~10 lines down), since it is not used until then.

Reply:
Fixed!